core/src/main/java/org/apache/iceberg/FileScanTaskParser.java

```diff
@@ -0,0 +1,130 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ExpressionParser;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.expressions.ResidualEvaluator;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.util.JsonUtil;
+
+public class FileScanTaskParser {
+  private static final String SCHEMA = "schema";
+  private static final String SPEC = "spec";
+  private static final String DATA_FILE = "data-file";
+  private static final String DELETE_FILES = "delete-files";
+  private static final String RESIDUAL = "residual-filter";
+
+  private FileScanTaskParser() {}
+
+  public static String toJson(FileScanTask fileScanTask) {
+    return JsonUtil.generate(
+        generator -> FileScanTaskParser.toJson(fileScanTask, generator), false);
+  }
+
+  private static void toJson(FileScanTask fileScanTask, JsonGenerator generator)
+      throws IOException {
+    Preconditions.checkArgument(fileScanTask != null, "File scan task cannot be null");
+    generator.writeStartObject();
+
+    generator.writeFieldName(SCHEMA);
+    SchemaParser.toJson(fileScanTask.schema(), generator);
+
+    generator.writeFieldName(SPEC);
+    PartitionSpec spec = fileScanTask.spec();
+    PartitionSpecParser.toJson(spec, generator);
+
+    if (fileScanTask.file() != null) {
+      generator.writeFieldName(DATA_FILE);
+      ContentFileParser.toJson(fileScanTask.file(), spec, generator);
+    }
+
+    if (fileScanTask.deletes() != null) {
+      generator.writeArrayFieldStart(DELETE_FILES);
+      for (DeleteFile deleteFile : fileScanTask.deletes()) {
+        ContentFileParser.toJson(deleteFile, spec, generator);
+      }
+      generator.writeEndArray();
+    }
+
+    if (fileScanTask.residual() != null) {
+      generator.writeFieldName(RESIDUAL);
+      ExpressionParser.toJson(fileScanTask.residual(), generator);
+    }
+
+    generator.writeEndObject();
+  }
+
+  public static FileScanTask fromJson(String json, boolean caseSensitive) {
+    Preconditions.checkArgument(json != null, "Cannot parse file scan task from null JSON string");
+    try {
+      JsonNode jsonNode = JsonUtil.mapper().readValue(json, JsonNode.class);
```

Comment:
I think this can be replaced with `JsonUtil.parse(json, node -> FileScanTaskParser.fromJsonNode(node, caseSensitive))`

and then the null check can also be moved down to `fromJsonNode`